src/views/TodoView.vue

```diff
@@ -0,0 +1,329 @@
+<script setup lang="ts">
+import TableItem from "../components/TableItem.vue";
+import { computed, onMounted, ref } from "vue";
+import { ResponseData, TaskInfo, debounce } from "../util/shared";
+import {
+  deletePost,
+  getPosts,
+  postPost,
+  updatePost,
+} from "../util/HttpRequests";
+import FlexCard from "../components/FlexCard.vue";
+import FlexList from "../components/FlexList.vue";
+import FlexText from "../components/FlexText.vue";
+import AddTodo from "../components/AddTodo.vue"
+import { useRouter } from "vue-router";
+import IconButton from "../components/IconButton.vue";
+
+const DEBOUNCE_TIME = 250;
+
+let oldGet: ResponseData[]; // Keep a backup to restore from
+const router = useRouter()
+
+const sortingOrder = ref<SORT_OPTIONS>("task_dsc");
+
+const todoData = ref<ResponseData[]>([]);
+
+const filterInput = ref("");
+const addText = ref("");
+
+type SORT_OPTIONS = "task_asc" | "task_dsc" | "complete" | "incomplete";
+
+const sortingOptions: { text: string; value: SORT_OPTIONS }[] = [
+  { text: "Tasks (dsc)", value: "task_dsc" },
+  { text: "Tasks (asc)", value: "task_asc" },
+  { text: "Complete", value: "complete" },
+  { text: "Incomplete", value: "incomplete" },
+];
+
+const todoFiltered = computed(() => {
+  return [...todoData.value].filter((value) => {
+    // Copy or we end up blowing the stack
+    return value.todoName
+      .toLowerCase()
+      .includes(filterInput.value.toLowerCase());
+  });
+});
+
+
+const todoSorted = computed(() => {
+
+  const taskSort = (a: ResponseData, b: ResponseData) => {
+    return a.todoName.localeCompare(b.todoName); // Thanks Ethan!
+  };
+
+  const completeSort = (a: ResponseData, b: ResponseData) => {
+    return ~~b.isComplete - ~~a.isComplete;
```

Comment:
Had to google this one... This is cute, but in code that other people have to read I would just use the Number constructor to be explicit about your intentions. And here's some other ways to convert booleans to numbers https://stackoverflow.com/a/66000126. If you were gonna pick a shorthand to avoid typing out Number I'd probably go with the unary plus operator or a ternary as it's a little more obvious

Reply:
`!!` is very common though so feel free to use that instead of the `Boolean` constructor

Reply:
Yeah I like the unary plus, ternary seems a bit overkill. I feel that the `+` is used commonly enough to convert to numbers that it should be readible enough.